File: src/main/resources/static/js/reservation/reservationParam.js
function ReservationParam() {

}

ReservationParam.prototype = {
    makeReservationParam: function (displayInfo) {
        const reservationParam = {
            displayInfoId: 0,
            prices: [],
            productId: 0,
            reservationEmail: "",
            reservationName: "",
            reservationTelephone: "",
            reservationYearMonthDay: ""
        };

        reservationParam.displayInfoId = displayInfo.displayInfoId;
        const priceNodes = document.querySelectorAll("input[name=price]");
        priceNodes.forEach((priceNode) => {
            reservationParam.prices.push(JSON.parse(priceNode.defaultValue));
        });
        reservationParam.productId = displayInfo.productId;
        reservationParam.reservationEmail = document.querySelector("#email").value;
        reservationParam.reservationName = document.querySelector("#name").value;
        reservationParam.reservationTelephone = document.querySelector("#tel").value;
        reservationParam.reservationYearMonthDay = TicketInfo.prototype.reservationYearMonthDay;

        return reservationParam
    },
    requestReservationParam: async function (reservationParam) {
        try {
            const url = "/api/reservations";
            const reservationResponse = await ajax("POST", url, reservationParam);
            window.location.replace("/");
        } catch (error) {
            console.log(error);
        }

    }
};

File: src/main/java/me/weekbelt/naverreservation/web/dto/reservation/ReservationParam.java
package me.weekbelt.naverreservation.web.dto.reservation;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@Getter
public class ReservationParam {

    private Long displayInfoId;
    private Long productId;

    private String reservationName;
    private String reservationTelephone;
    private String reservationEmail;
    private String reservationYearMonthDay;

    private List<ReservationPriceDto> prices;

    @Builder
    public ReservationParam(Long displayInfoId, Long productId, String reservationName, String reservationTelephone, String reservationEmail, String reservationYearMonthDay, List<ReservationPriceDto> prices) {
        this.displayInfoId = displayInfoId;
        this.productId = productId;
        this.reservationName = reservationName;
        this.reservationTelephone = reservationTelephone;
        this.reservationEmail = reservationEmail;
        this.reservationYearMonthDay = reservationYearMonthDay;
        this.prices = prices;
    }

}
